src/descriptors/power_system_structs.json

```diff
@@ -1860,6 +1921,181 @@
+                "name": "activepower",
+                "null_value": "0.0",
+                "data_type": "Float64",
+                "valid_range": "activepowerlimits",
+                "validation_action": "warn"
+            },
+            {
+                "name": "reactivepower",
+                "null_value": "0.0",
+                "data_type": "Float64",
+                "valid_range": "reactivepowerlimits",
+                "validation_action": "warn"
+            },
+            {
+                "name": "rating",
+                "comment": "Thermal limited MVA Power Output of the unit. <= Capacity",
+                "null_value": "0.0",
+                "data_type": "Float64",
+                "valid_range": {
+                    "min": 0,
+                    "max": null
+                },
+                "validation_action": "error"
+            },
+            {
+                "name": "primemover",
+                "comment": "PrimeMover Technology according to EIA 923",
+                "null_value": "PrimeMovers.OT",
+                "data_type": "PrimeMovers.PrimeMover"
+            },
+            {
+                "name": "fuel",
+                "comment": "PrimeMover Fuel according to EIA 923",
+                "null_value": "ThermalFuels.OTHER",
+                "data_type": "ThermalFuels.ThermalFuel"
+            },
+            {
+                "name": "activepowerlimits",
+                "null_value": "(min=0.0, max=0.0)",
+                "data_type": "Min_Max"
+            },
+            {
+                "name": "reactivepowerlimits",
+                "null_value": "nothing",
+                "data_type": "Union{Nothing, Min_Max}"
+            },
+            {
+                "name": "ramplimits",
+                "null_value": "nothing",
+                "data_type": "Union{Nothing, NamedTuple{(:up, :down), Tuple{Float64, Float64}}}",
+                "valid_range": {
+                    "min": 0,
+                    "max": null
+                },
+                "validation_action": "error"
+            },
+            {
+                "name": "powertrajectory",
+                "null_value": "nothing",
+                "data_type": "Union{Nothing, NamedTuple{(:startup, :shutdown), Tuple{Float64, Float64}}}",
+                "valid_range": {
+                    "min": 0,
+                    "max": null
+                },
+                "validation_action": "error"
+            },
+            {
+                "name": "timelimits",
+                "comment": "Minimum up and Minimum down time limits in hours",
+                "null_value": "nothing",
+                "data_type": "Union{Nothing, NamedTuple{(:up, :down), Tuple{Float64, Float64}}}",
+                "valid_range": {
+                    "min": 0,
+                    "max": null
+                },
+                "validation_action": "error"
+            },
+            {
+                "name": "starttimelimits",
```

Comment:
Can you use `snake_case` for all new fields?  These are hard to read.

Reply:
yup